.github/workflows/build_lint.yml

```diff
@@ -46,30 +46,11 @@ jobs:
       uses: actions/setup-python@v5
       with:
         python-version: ${{ matrix.python-version }}
-    - if: ${{ runner.os == 'Windows' }}
-      # This is needed so that restoring cache on Windows is fast.
-      # See until https://github.com/actions/cache/issues/752 is resolved.
-      name: Use GNU tar
-      shell: cmd
-      run: |
-        echo "Adding GNU tar to PATH"
-        echo C:\Program Files\Git\usr\bin>>"%GITHUB_PATH%"
-    - uses: actions/cache@v4
-      with:
-        path: ${{ env.pythonLocation }}
-        key: ${{ runner.os }}-pip-${{ env.pythonLocation }}-${{ hashFiles('setup.py') }}
-    - name: Install dependencies
-      # Only if the cache misses
-      # Based on https://github.com/pypa/pip/issues/8049#issuecomment-633845028
-      # read_requirements.py should be removed once
-      # https://github.com/pypa/pip/issues/11440 is resolved.
-      if: steps.cache.outputs.cache-hit != 'true'
-      run: |
-        pip install toml
-        python tests/read_requirements.py > requirements.txt
-        pip install -r requirements.txt
+        cache: 'pip'
+    - name: Install uv
+      run: pip install uv
     - name: Install Mesa
-      run: pip install --no-deps .
+      run: uv pip install --system -e .[dev]
```

Comment:
Is the edible install needed here?

Reply:
There are 2 options as described in https://github.com/astral-sh/uv/issues/1945.

Reply:
I saw this linked from uv -- just as a heads up, this is supported as of `v0.1.24`. You can `uv pip install .` directly, without using an editable.

Reply:
@charliermarsh thank you for the awesome tool! I have amended the PR with `uv pip install --system .[dev]`.